experimental/dds/tree2/src/test/feature-libraries/optional-field/optionalChangeRebaser.spec.ts

```diff
@@ -189,6 +206,44 @@ function compose(
 
 type OptionalFieldTestState = FieldStateTree<string | undefined, OptionalChangeset>;
 
+function computeChildChangeInputContext(inputState: OptionalFieldTestState): number[] {
+	// This is effectively a filter of the intentions from all edits such that it only includes
+	// intentions for edits which modify the same child as the final one in the changeset.
+	// Note: this takes a dependency on the fact that `generateChildStates` doesn't set matching string
+	// content for what are meant to represent different nodes.
+	const states: OptionalFieldTestState[] = [];
+	for (
+		let current: OptionalFieldTestState | undefined = inputState;
+		current !== undefined;
+		current = current.parent
+	) {
+		states.push(current);
+	}
```

Comment:
Looks like this part of the logic could just reuse `getSequentialEdits`, or, even better, take in the array as parameter instead of recomputing it here (in which case you would iterate it backwards instead of reversing it).